From 0c0928c2fba54c7b95dd3c9cf0436846ded57a3d Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Sun, 28 Jun 2020 19:44:20 +0200
Subject: [PATCH 31/51] cht-bsw-rt5672: Boost ADC volume a bit

Both the DMICs and the headset mic are quite soft, even with the 20 dB
IN1 boost we already congfigure for the headset.

This commit sets the ADC Boost gain (range 0-3) to 1 = 12 dB, which makes
the volume a lot better. The next step of the ADC Boost gain setting is
24 dB which is a bit too much.

This has been tested with both the builtin DMIC and with a headset mic
on the following devices:

Lenovo Miix 2 10
Lenovo Thinkpad 8
Lenovo Thinkpad 10 (gen 1)

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---
 ucm2/codecs/rt5672/EnableSeq.conf | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/ucm2/codecs/rt5672/EnableSeq.conf b/ucm2/codecs/rt5672/EnableSeq.conf
index a19a98291b8c..ddfb1c7b6c72 100644
--- a/ucm2/codecs/rt5672/EnableSeq.conf
+++ b/ucm2/codecs/rt5672/EnableSeq.conf
@@ -30,6 +30,11 @@ EnableSequence [
 	cset "name='ADC Capture Volume' 47"
 	cset "name='ADC Capture Switch' on"
 
+	# Both the DMICs and the headset mic (even with the 20 dB boost below)
+	# are quite soft, set the ADC Boost gain (range 0-3) to 1 = 12 dB the
+	# next step is 24 dB which is a bit too much
+	cset "name='STO1 ADC Boost Gain Volume' 1"
+
 	# Headset mic is quite soft, boost it a bit, 1 = 20dB which is the first
 	# available boost step
 	cset "name='IN1 Boost Volume' 1"
-- 
2.16.4

